### PR TITLE
fix: new field: Started in the device deployment struct.

### PR DESCRIPTION
**Pull request by merlin-northern**
Introduce a new field that denotes a time when device actually asked for a deployment.

Changelog: Title
Ticket: MEN-6793

### CODE REVIEW

**Comment by merlin-northern** [comment]
@tranchitella I need initial review on this one. does it sound like a reasonable approach?

**Review by tranchitella — CHANGES_REQUESTED** [review]
I'm not sure that this is the right approach. By the way, there is something strange in the management of the `Created`, see here:
https://github.com/mendersoftware/deployments/blob/master/app/app.go#L1413

However, when we initialize the device deployment, we set `Created` to now:
https://github.com/mendersoftware/deployments/blob/master/model/device_deployment.go#L269

Your implementation has a drawback: it will issue a new (update) query when the deployment starts. I'm wondering whether this needs to be effectively done only if the deployment was previously pending. If we are inserting the deployment, we can set it straight away in the Insert query.

Additionally, you need to update API docs.

**Comment by merlin-northern** [comment]
> I'm not sure that this is the right approach. By the way, there is something strange in the management of the `Created`, see here: https://github.com/mendersoftware/deployments/blob/master/app/app.go#L1413
> 
> However, when we initialize the device deployment, we set `Created` to now: https://github.com/mendersoftware/deployments/blob/master/model/device_deployment.go#L269
> 
> Your implementation has a drawback: it will issue a new (update) query when the deployment starts. I'm wondering whether this needs to be effectively done only if the deployment was previously pending. If we are inserting the deployment, we can set it straight away in the Insert query.
> 
> Additionally, you need to update API docs.

thank you. fixed. I will try to take a look at the `Created` field, but this one is about started, so maybe when we change GUI to show it next to the `Started` label we do not have to fix `Created` in this PR? if you consider checking `Created` field handling as a part of this task, let me know I will regroup.

File: store/mongo/datastore_mongo.go [hunk 1]
```diff
@@ -1385,6 +1386,11 @@ func (db *DataStoreMongo) InsertDeviceDeployment(
 	database := db.client.Database(mstore.DbFromContext(ctx, DatabaseName))
 	c := database.Collection(CollectionDevices)
 
+	if deviceDeployment.Status == model.DeviceDeploymentStatusPending {
+		startedTime := time.Now().UTC()
+		deviceDeployment.Started = &startedTime
+	}
```
[the remark this answers is not in the document]
Comment (merlin-northern) [reply store/mongo/datastore_mongo.go:1392]: wow. I am sorry. it is a lesson to me: never send code from PTO with your head somewhere completely else. I am fixing.

[the remark this answers is not in the document]
Comment (merlin-northern) [reply store/mongo/datastore_mongo.go]: I will just take my time, testing properly, and fix. sorry for wasting your time.

[the remark this answers is not in the document]
Comment (merlin-northern) [reply store/mongo/datastore_mongo.go]: I will just take my time, testing properly, and fix. sorry for wasting your time.

**Review by tranchitella — APPROVED** [review]

**Pull request merged**